File: mobile_through_project_final/src/main/java/com/ssafy/cafe/model/dto/CouponDetail.java
package com.ssafy.cafe.model.dto;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class CouponDetail {
    Integer couponId;
    Integer couponTypeId;
    String couponName;
    Integer discountRate;
    String userId;
    Boolean isUse;
    
    @Builder
    public CouponDetail(Integer couponId, Integer couponTypeId, String couponName, Integer discountRate, String userId,
            Boolean isUse) {
        super();
        this.couponId = couponId;
        this.couponTypeId = couponTypeId;
        this.couponName = couponName;
        this.discountRate = discountRate;
        this.userId = userId;
        this.isUse = isUse;
    }
}

File: app/src/main/java/com/ssafy/cobaltcoffee/dto/CouponDetail.kt
package com.ssafy.cobaltcoffee.dto

data class CouponDetail(
    val couponId: Int,
    val couponTypeId: Int,
    val couponName: String,
    val discountRate: Int,
    val userId: String,
    val isUse: Boolean
) {
    constructor() : this(0, 0, "", 0, "", false)
}